File: abc_spec/code_samples/Java/webhooks@{id}/patch.java
// For more information please refer to https://github.com/checkout/checkout-sdk-java
import com.checkout.CheckoutApiException;
import com.checkout.CheckoutArgumentException;
import com.checkout.CheckoutAuthorizationException;
import com.checkout.CheckoutSdk;
import com.checkout.Environment;
import com.checkout.previous.CheckoutApi;
import com.checkout.webhooks.previous.WebhookRequest;
import com.checkout.webhooks.previous.WebhookResponse;

CheckoutApi api = CheckoutSdk
     .builder()
     .previous()
     .staticKeys()
     .secretKey("secret_key")
     .environment(Environment.SANDBOX) // or Environment.PRODUCTION
     .build();

WebhookRequest webhookRequest = WebhookRequest.builder()
    .url("https://docs.checkout.com/webhooks/updated")
    .eventTypes(Arrays.asList("payment_captured", "payment_approved", "payment_declined"))
    .build();

try {
    WebhookResponse response = api.webhooksClient().updateWebhook("webhookId", webhookRequest).get();
} catch (CheckoutApiException e) {
    // API error
    String requestId = e.getRequestId();
    int statusCode = e.getHttpStatusCode();
    Map<String, Object> errorDetails = e.getErrorDetails();
} catch (CheckoutArgumentException e) {
    // Bad arguments
} catch (CheckoutAuthorizationException e) {
    // Invalid authorization
}

File: abc_spec/code_samples/PHP/webhooks@{id}/patch.php
<?php
//For more information please refer to https://github.com/checkout/checkout-sdk-php

use Checkout\\CheckoutApiException;
use Checkout\\CheckoutAuthorizationException;
use Checkout\\CheckoutSdk;
use Checkout\\Environment;
use Checkout\\Webhooks\\Previous\\WebhookRequest;

$api = CheckoutSdk::builder()
    ->previous()
    ->staticKeys()
    ->environment(Environment::sandbox())
    ->secretKey("secret_key")
    ->build();

$request = new WebhookRequest();
$request->url = "https://test.checkout.com/webhooks/changed";
$request->headers = [];
$request->content_type = "json";
$request->event_types = array("source_updated");
$request->active = true;

try {
    $response = $api->getWebhooksClient()->patchWebhook("webhookId", $request);
} catch (CheckoutApiException $e) {
    // API error
    $error_details = $e->error_details;
    $http_status_code = isset($e->http_metadata) ? $e->http_metadata->getStatusCode() : null;
} catch (CheckoutAuthorizationException $e) {
    // Bad Invalid authorization
}
